typing/typedecl.ml

```diff
@@ -2305,6 +2314,9 @@ let report_error ~loc = function
          it should not occur deeply into its type."
         Style.inline_code
         (match kind with Unboxed -> "@unboxed" | Untagged -> "@untagged")
+  | Type_cannot_be_external ->
+      Location.errorf ~loc
+        "This type cannot be used to annotate an external function."
```

Comment:
This is a bit sibylline for an error message. At least, the error message should say which type is not supported.